### PR TITLE
[WASH-1045] Identity login locks tests

### PR DESCRIPTION
**Pull request by Mrhea**

### CODE REVIEW

File: identityClient/api.go [hunk 1]
```diff
@@ -226,6 +226,26 @@ type InternalIdentityLoginResponse struct {
...
+// InternalIdentityStatusResponse wraps an internal Identity
+// status context for realm identities.
+type InternalIdentityStatusResponse struct {
+	Locked bool `json:"locked"` // Whether the Identity's account is currently locked for logging into a realm.
+}
+
+// InternalIdentityStatusUserIdRequest wraps the parameters
+// needed to determine the status of an Identity from its Identity User ID.
+type InternalIdentityStatusUserIdRequest struct {
+	RealmName string    // The name of the realm the Identity is a member of.
```
Comment (galxy25) [inline identityClient/api.go:238]: Reminder to use RealmDomainName everywhere as thats the internal service identifier (vs the realm name which could be upper or mixed cased)

Comment (Mrhea) [reply identityClient/api.go:238]: Ah yes, I need to make sure I clarify that for myself. I will go through all my PRs and update this naming

**Review by galxy25 — APPROVED** [review]

File: identityClient/api.go [hunk 2]
```diff
@@ -226,6 +226,26 @@ type InternalIdentityLoginResponse struct {
...
+// InternalIdentityStatusResponse wraps an internal Identity
+// status context for realm identities.
+type InternalIdentityStatusResponse struct {
+	Locked bool `json:"locked"` // Whether the Identity's account is currently locked for logging into a realm.
+}
+
+// InternalIdentityStatusUserIdRequest wraps the parameters
+// needed to determine the status of an Identity from its Identity User ID.
+type InternalIdentityStatusUserIdRequest struct {
+	RealmDomainName string    // The name of the realm the Identity is a member of.
```
Comment (Mrhea) [inline identityClient/api.go:238]: `RealmName` --> `RealmDomainName`

File: identityClient/identityClient.go [hunk 3]
```diff
@@ -755,6 +755,28 @@ func (c *E3dbIdentityClient) InternalDeleteLDAPCache(ctx context.Context, realmN
...
+	if err != nil {
+		return identityStatus, err
+	}
+	err = e3dbClients.MakeSignedServiceCall(ctx, c.requester, req, c.SigningKeys, c.ClientID, &identityStatus)
+	return identityStatus, err
+}
+
+func (c *E3dbIdentityClient) InternalIdentityStatusByStorageClientId(ctx context.Context, params InternalIdentityStatusStorageClientIdRequest) (*InternalIdentityStatusResponse, error) {
+	var identityStatus *InternalIdentityStatusResponse
+	path := c.Host + internalIdentityServiceBasePath + "/" + realmResourceName + "/" + params.RealmDomainName + "/identity_status_by_storage_client_id/" + params.StorageClientID.String()
```
Comment (Mrhea) [inline identityClient/identityClient.go:771]: `params.RealmName` --> `params.RealmDomainName`

File: identityClient/identityClient_internal_test.go [hunk 4]
```diff
@@ -56,6 +56,137 @@ var (
...
+	anonConfig := e3dbClients.ClientConfig{
+		Host: e3dbIdentityHost,
+	}
+	anonClient := New(anonConfig)
+	identity, err := anonClient.RegisterIdentity(testContext, registerParams)
+	if err != nil {
+		t.Errorf("RegisterIdentity Error: %+v\n", err)
+	}
+	reqParams := InternalIdentityStatusUserIdRequest{
+		RealmDomainName: realm.Domain,
```
Comment (Mrhea) [inline identityClient/identityClient_internal_test.go:116]: Using `realm.Domain` instead of `realm.Name`

**Review by pirtleshell — APPROVED** [review]

File: identityClient/api.go [hunk 5]
```diff
@@ -226,6 +226,26 @@ type InternalIdentityLoginResponse struct {
...
+	UserID          uuid.UUID // The ID of the Identity's user.
+}
+
+// InternalIdentityStatusStorageClientIdRequest wraps the parameters
+// needed to determine the status of an Identtiy from its Storage Client ID
+type InternalIdentityStatusStorageClientIdRequest struct {
+	RealmDomainName string    // The name of the realm the Identity is a member of.
```
Comment (rosasck) [inline identityClient/api.go:245]: would prefer RealmDomain to follow what we currently do

**Review by rosasck — APPROVED** [review]
To be consistent lets use RealmDomain

File: identityClient/api.go (hunk 1 above)
Comment (rosasck) [reply identityClient/api.go]: @galxy25 Should we use RealmDomain to keep consistent with other structs

**Pull request merged**